From 95fb37fc68cb8a760fcbb048b2fc4dba4bca30a6 Mon Sep 17 00:00:00 2001
From: zhengq <qi.zheng@intel.com>
Date: Thu, 7 May 2015 00:02:00 +0800
Subject: [PATCH] PMIC: disable power button support.

For dollar cove pmic, there is power button support in default.
Thus we have two input events for reporting power key event.
One is "gpio-keys", the other is "dollar_cove_power_button".
Then one time power button press has two times power key event
reported. This will cause the system wake-up and sleep again
with just one time power button press.
Add a definition PMIC_DC_PWR_BTN_EN in pmic dc driver.
If need the pmic support power button event report, define it.
Otherwise, like in this case, the power button event report is
not supported.

Change-Id: Iddf4fee164ccabbb9bdc7c4c770429cf64ab3604
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1558
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-30984
Signed-off-by: zhengq <qi.zheng@intel.com>
---
 drivers/mfd/intel_soc_pmic_dc.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/mfd/intel_soc_pmic_dc.c b/drivers/mfd/intel_soc_pmic_dc.c
index b1e0b5c0f0d0..a61788136c0f 100644
--- a/drivers/mfd/intel_soc_pmic_dc.c
+++ b/drivers/mfd/intel_soc_pmic_dc.c
@@ -39,6 +39,9 @@
 #define GPIO_OTG_VBUS_INDEX	2
 #define FPO0_USB_COMP_OFFSET	0x01
 
+/*If need PMIC support power button event report, define it*/
+/*#define PMIC_DC_PWR_BTN_EN*/
+
 enum {
 	VBUS_FALLING_IRQ = 2,
 	VBUS_RISING_IRQ,
@@ -81,6 +84,7 @@ enum {
 	BC_USB_CHNG_IRQ,
 };
 
+#ifdef PMIC_DC_PWR_BTN_EN
 static struct resource power_button_resources[] = {
 	{
 		.start	= POKN_IRQ,
@@ -93,7 +97,7 @@ static struct resource power_button_resources[] = {
 		.flags	= IORESOURCE_IRQ,
 	},
 };
-
+#endif
 
 static struct resource gpio_resources[] = {
 	{
@@ -228,12 +232,14 @@ static struct mfd_cell dollar_cove_dev[] = {
 		.num_resources = ARRAY_SIZE(gpio_resources),
 		.resources = gpio_resources,
 	},
+#ifdef PMIC_DC_PWR_BTN_EN
 	{
 		.name = "dollar_cove_power_button",
 		.id = 0,
 		.num_resources = ARRAY_SIZE(power_button_resources),
 		.resources = power_button_resources,
 	},
+#endif
 	{
 		.name = "dollar_cove_pwrsrc",
 		.id = 0,
-- 
1.9.1

